src/test/java/com/example/SpringExample/EmployeeServiceImplTest.java

```diff
@@ -0,0 +1,95 @@
+package com.example.SpringExample;
+
+import com.example.SpringExample.entity.Employee;
+import com.example.SpringExample.repository.EmployeeRepository;
+import com.example.SpringExample.service.EmployeeServiceImpl;
+import com.example.SpringExample.service.ResourceAlreadyExistsException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+public class EmployeeServiceImplTest {
+
+    @Mock
+    private EmployeeRepository employeeRepository;
+
+    @InjectMocks
+    private EmployeeServiceImpl employeeService;
+
+    Employee employee;
+
+    @BeforeEach
+    void setup(){
+        MockitoAnnotations.openMocks(this);
+        employee = new Employee(1,"Madhu", "Shekhawat","mkshekhawat47@gmail.com");
+    }
+
+    @Test
+    void testSaveEmployee_NewEmployee(){
+        when(employeeRepository.findByEmail(anyString())).thenReturn(Optional.empty());
+        employeeService.saveEmployee(employee);
+        ArgumentCaptor<Employee> captor = ArgumentCaptor.forClass(Employee.class);
+        verify(employeeRepository).save(captor.capture());
+        Employee capturedEmployee = captor.getValue();
+        Assertions.assertEquals("mkshekhawat47@gmail.com", capturedEmployee.getEmail());
```

Comment:
Instead of checking if the two objects have the same id, can we compare the objects itself?